### ISSUE TITLE
Pip install plantcv fails at scikit-image: Error -3 while decompressing: invalid code lengths set

### ISSUE THREAD

**Issue by bigFin** [danforthcenter/plantcv#428]
System: Windows 10, Python 3.7, virtualenv
Issue: Pip install plantcv fails with error 3 while decompressing zipfile.py
Output:
`pip install plantcv
Collecting plantcv
  Using cached https://files.pythonhosted.org/packages/b5/af/e95fb1c119ab91da77c2b4d52df54b1a1e59dc7a24f4f9e80ed3379d7653/plantcv-3.5.0-py3-none-any.whl
Collecting numpy>=1.11 (from plantcv)
  Using cached https://files.pythonhosted.org/packages/f9/3f/d75fc983cc420b2acb5fae446b950e2dc9e5395a79fa76859d2528352d2c/numpy-1.17.0-cp37-cp37m-win32.whl
Collecting matplotlib>=1.5 (from plantcv)
  Using cached https://files.pythonhosted.org/packages/71/13/0720e50bd8988299137fd7e936e4d494b45a473c5fe70d72cd6c1bd79163/matplotlib-3.1.1-cp37-cp37m-win32.whl
Collecting pandas (from plantcv)
  Using cached https://files.pythonhosted.org/packages/22/1e/9eebae0a7e077301e54d73c6c510ecce6b648b843ed6b6d4eeddcd74022c/pandas-0.25.0-cp37-cp37m-win32.whl
Collecting scikit-image<0.15 (from plantcv)
  Using cached https://files.pythonhosted.org/packages/bd/5d/df4b0e528bd4404c9e27598da2b1c9d367c64e7a4e76e1a7cbf7c39a7b70/scikit_image-0.14.3-cp37-none-win32.whl
ERROR: Exception:
Traceback (most recent call last):
  File "c:\.venv\lib\site-packages\pip\_internal\cli\base_command.py", line 188, in main
    status = self.run(options, args)
  File "c:\.venv\lib\site-packages\pip\_internal\commands\install.py", line 345, in run
    resolver.resolve(requirement_set)
  File "c:\.venv\lib\site-packages\pip\_internal\legacy_resolve.py", line 196, in resolve
    self._resolve_one(requirement_set, req)
  File "c:\.venv\lib\site-packages\pip\_internal\legacy_resolve.py", line 359, in _resolve_one
    abstract_dist = self._get_abstract_dist_for(req_to_install)
  File "c:\.venv\lib\site-packages\pip\_internal\legacy_resolve.py", line 307, in _get_abstract_dist_for
    self.require_hashes
  File "c:\.venv\lib\site-packages\pip\_internal\operations\prepare.py", line 199, in prepare_linked_requirement
    progress_bar=self.progress_bar
  File "c:\.venv\lib\site-packages\pip\_internal\download.py", line 1064, in unpack_url
    progress_bar=progress_bar
  File "c:\.venv\lib\site-packages\pip\_internal\download.py", line 928, in unpack_http_url
    unpack_file(from_path, location, content_type, link)
  File "c:\.venv\lib\site-packages\pip\_internal\utils\misc.py", line 735, in unpack_file
    flatten=not filename.endswith('.whl')
  File "c:\.venv\lib\site-packages\pip\_internal\utils\misc.py", line 631, in unzip_file
    shutil.copyfileobj(fp, destfp)
  File "C:\Users\User\AppData\Local\Programs\Python\Python37-32\lib\shutil.py", line 79, in copyfileobj
    buf = fsrc.read(length)
  File "C:\Users\User\AppData\Local\Programs\Python\Python37-32\lib\zipfile.py", line 899, in read
    data = self._read1(n)
  File "C:\Users\User\AppData\Local\Programs\Python\Python37-32\lib\zipfile.py", line 975, in _read1
    data = self._decompressor.decompress(data, n)
zlib.error: Error -3 while decompressing data: invalid code lengths set`

**Comment by bigFin** [comment]
Installing plantcv from git failed similarly.
`
Searching for scikit-image<0.15
Reading https://pypi.org/simple/scikit-image/
Downloading https://files.pythonhosted.org/packages/bd/5d/df4b0e528bd4404c9e27598da2b1c9d367c64e7a4e76e1a7cbf7c39a7b70/scikit_image-0.14.3-cp37-none-win32.whl#sha256=b7b5605c7737073c1bdffbfea1f6a257dd6e96ab109fc37f43a7bd3d1d581e2a
Best match: scikit-image 0.14.3
Processing scikit_image-0.14.3-cp37-none-win32.whl
Installing scikit_image-0.14.3-cp37-none-win32.whl to c:\users\User\onedrive - canntx life sciences\lab_g\image_analysis\code\plantcv_canntx_1\.venv\lib\site-packages
Traceback (most recent call last):
  File "setup.py", line 117, in <module>
    scripts=["plantcv-train.py", "plantcv-utils.py", "plantcv-workflow.py"]
  File "C:\Users\User\OneDrive - Canntx Life Sciences\LAB_G\Image_analysis\Code\PlantCV_CannTx_1\.venv\lib\site-packages\setuptools\__init__.py", line 145, in setup
    return distutils.core.setup(**attrs)
  File "C:\Users\User\AppData\Local\Programs\Python\Python37-32\lib\distutils\core.py", line 148, in setup
    dist.run_commands()
  File "C:\Users\User\AppData\Local\Programs\Python\Python37-32\lib\distutils\dist.py", line 966, in run_commands
    self.run_command(cmd)
  File "C:\Users\User\AppData\Local\Programs\Python\Python37-32\lib\distutils\dist.py", line 985, in run_command
    cmd_obj.run()
  File "C:\Users\User\OneDrive - Canntx Life Sciences\LAB_G\Image_analysis\Code\PlantCV_CannTx_1\.venv\lib\site-packages\setuptools\command\install.py", line 67, in run
    self.do_egg_install()
  File "C:\Users\User\OneDrive - Canntx Life Sciences\LAB_G\Image_analysis\Code\PlantCV_CannTx_1\.venv\lib\site-packages\setuptools\command\install.py", line 117, in do_egg_install
    cmd.run()
  File "C:\Users\User\OneDrive - Canntx Life Sciences\LAB_G\Image_analysis\Code\PlantCV_CannTx_1\.venv\lib\site-packages\setuptools\command\easy_install.py", line 418, in run
    self.easy_install(spec, not self.no_deps)
  File "C:\Users\User\OneDrive - Canntx Life Sciences\LAB_G\Image_analysis\Code\PlantCV_CannTx_1\.venv\lib\site-packages\setuptools\command\easy_install.py", line 660, in easy_install
    return self.install_item(None, spec, tmpdir, deps, True)
  File "C:\Users\User\OneDrive - Canntx Life Sciences\LAB_G\Image_analysis\Code\PlantCV_CannTx_1\.venv\lib\site-packages\setuptools\command\easy_install.py", line 707, in install_item
    self.process_distribution(spec, dist, deps)
  File "C:\Users\User\OneDrive - Canntx Life Sciences\LAB_G\Image_analysis\Code\PlantCV_CannTx_1\.venv\lib\site-packages\setuptools\command\easy_install.py", line 752, in process_distribution
    [requirement], self.local_index, self.easy_install
  File "C:\Users\User\OneDrive - Canntx Life Sciences\LAB_G\Image_analysis\Code\PlantCV_CannTx_1\.venv\lib\site-packages\pkg_resources\__init__.py", line 782, in resolve
    replace_conflicting=replace_conflicting
  File "C:\Users\User\OneDrive - Canntx Life Sciences\LAB_G\Image_analysis\Code\PlantCV_CannTx_1\.venv\lib\site-packages\pkg_resources\__init__.py", line 1065, in best_match
    return self.obtain(req, installer)
  File "C:\Users\User\OneDrive - Canntx Life Sciences\LAB_G\Image_analysis\Code\PlantCV_CannTx_1\.venv\lib\site-packages\pkg_resources\__init__.py", line 1077, in obtain
    return installer(requirement)
  File "C:\Users\User\OneDrive - Canntx Life Sciences\LAB_G\Image_analysis\Code\PlantCV_CannTx_1\.venv\lib\site-packages\setuptools\command\easy_install.py", line 679, in easy_install
    return self.install_item(spec, dist.location, tmpdir, deps)
  File "C:\Users\User\OneDrive - Canntx Life Sciences\LAB_G\Image_analysis\Code\PlantCV_CannTx_1\.venv\lib\site-packages\setuptools\command\easy_install.py", line 705, in install_item
    dists = self.install_eggs(spec, download, tmpdir)
  File "C:\Users\User\OneDrive - Canntx Life Sciences\LAB_G\Image_analysis\Code\PlantCV_CannTx_1\.venv\lib\site-packages\setuptools\command\easy_install.py", line 855, in install_eggs
    return [self.install_wheel(dist_filename, tmpdir)]
  File "C:\Users\User\OneDrive - Canntx Life Sciences\LAB_G\Image_analysis\Code\PlantCV_CannTx_1\.venv\lib\site-packages\setuptools\command\easy_install.py", line 1073, in install_wheel
    os.path.dirname(destination)
  File "C:\Users\User\AppData\Local\Programs\Python\Python37-32\lib\distutils\cmd.py", line 335, in execute
    util.execute(func, args, msg, dry_run=self.dry_run)
  File "C:\Users\User\AppData\Local\Programs\Python\Python37-32\lib\distutils\util.py", line 291, in execute
    func(*args)
  File "C:\Users\User\OneDrive - Canntx Life Sciences\LAB_G\Image_analysis\Code\PlantCV_CannTx_1\.venv\lib\site-packages\setuptools\wheel.py", line 101, in install_as_egg
    self._install_as_egg(destination_eggdir, zf)
  File "C:\Users\User\OneDrive - Canntx Life Sciences\LAB_G\Image_analysis\Code\PlantCV_CannTx_1\.venv\lib\site-packages\setuptools\wheel.py", line 109, in _install_as_egg
    self._convert_metadata(zf, destination_eggdir, dist_info, egg_info)
  File "C:\Users\User\OneDrive - Canntx Life Sciences\LAB_G\Image_analysis\Code\PlantCV_CannTx_1\.venv\lib\site-packages\setuptools\wheel.py", line 131, in _convert_metadata
    zf.extractall(destination_eggdir)
  File "C:\Users\User\AppData\Local\Programs\Python\Python37-32\lib\zipfile.py", line 1619, in extractall
    self._extract_member(zipinfo, path, pwd)
  File "C:\Users\User\AppData\Local\Programs\Python\Python37-32\lib\zipfile.py", line 1674, in _extract_member
    shutil.copyfileobj(source, target)
  File "C:\Users\User\AppData\Local\Programs\Python\Python37-32\lib\shutil.py", line 79, in copyfileobj
    buf = fsrc.read(length)
  File "C:\Users\User\AppData\Local\Programs\Python\Python37-32\lib\zipfile.py", line 899, in read
    data = self._read1(n)
  File "C:\Users\User\AppData\Local\Programs\Python\Python37-32\lib\zipfile.py", line 975, in _read1
    data = self._decompressor.decompress(data, n)
zlib.error: Error -3 while decompressing data: invalid code lengths set
`

**Comment by bigFin** [comment]
Installation from a git clone fails similarly.

`
Searching for scikit-image<0.15
Reading https://pypi.org/simple/scikit-image/
Downloading https://files.pythonhosted.org/packages/bd/5d/df4b0e528bd4404c9e27598da2b1c9d367c64e7a4e76e1a7cbf7c39a7b70/scikit_image-0.14.3-cp37-none-win32.whl#sha256=b7b5605c7737073c1bdffbfea1f6a257dd6e96ab109fc37f43a7bd3d1d581e2a
Best match: scikit-image 0.14.3
Processing scikit_image-0.14.3-cp37-none-win32.whl
Installing scikit_image-0.14.3-cp37-none-win32.whl to c:\users\User\.venv\lib\site-packages
Traceback (most recent call last):
  File "setup.py", line 117, in <module>
    scripts=["plantcv-train.py", "plantcv-utils.py", "plantcv-workflow.py"]
  File "C:\Users\User\.venv\lib\site-packages\setuptools\__init__.py", line 145, in setup
    return distutils.core.setup(**attrs)
  File "C:\Users\User\AppData\Local\Programs\Python\Python37-32\lib\distutils\core.py", line 148, in setup
    dist.run_commands()
  File "C:\Users\User\AppData\Local\Programs\Python\Python37-32\lib\distutils\dist.py", line 966, in run_commands
    self.run_command(cmd)
  File "C:\Users\User\AppData\Local\Programs\Python\Python37-32\lib\distutils\dist.py", line 985, in run_command
    cmd_obj.run()
  File "C:\Users\User\.venv\lib\site-packages\setuptools\command\install.py", line 67, in run
    self.do_egg_install()
  File "C:\Users\User\.venv\lib\site-packages\setuptools\command\install.py", line 117, in do_egg_install
    cmd.run()
  File "C:\Users\User\.venv\lib\site-packages\setuptools\command\easy_install.py", line 418, in run
    self.easy_install(spec, not self.no_deps)
  File "C:\Users\User\.venv\lib\site-packages\setuptools\command\easy_install.py", line 660, in easy_install
    return self.install_item(None, spec, tmpdir, deps, True)
  File "C:\Users\User\.venv\lib\site-packages\setuptools\command\easy_install.py", line 707, in install_item
    self.process_distribution(spec, dist, deps)
  File "C:\Users\User\.venv\lib\site-packages\setuptools\command\easy_install.py", line 752, in process_distribution
    [requirement], self.local_index, self.easy_install
  File "C:\Users\User\.venv\lib\site-packages\pkg_resources\__init__.py", line 782, in resolve
    replace_conflicting=replace_conflicting
  File "C:\Users\User\.venv\lib\site-packages\pkg_resources\__init__.py", line 1065, in best_match
    return self.obtain(req, installer)
  File "C:\Users\User\.venv\lib\site-packages\pkg_resources\__init__.py", line 1077, in obtain
    return installer(requirement)
  File "C:\Users\User\.venv\lib\site-packages\setuptools\command\easy_install.py", line 679, in easy_install
    return self.install_item(spec, dist.location, tmpdir, deps)
  File "C:\Users\User\.venv\lib\site-packages\setuptools\command\easy_install.py", line 705, in install_item
    dists = self.install_eggs(spec, download, tmpdir)
  File "C:\Users\User\.venv\lib\site-packages\setuptools\command\easy_install.py", line 855, in install_eggs
    return [self.install_wheel(dist_filename, tmpdir)]
  File "C:\Users\User\.venv\lib\site-packages\setuptools\command\easy_install.py", line 1073, in install_wheel
    os.path.dirname(destination)
  File "C:\Users\User\AppData\Local\Programs\Python\Python37-32\lib\distutils\cmd.py", line 335, in execute
    util.execute(func, args, msg, dry_run=self.dry_run)
  File "C:\Users\User\AppData\Local\Programs\Python\Python37-32\lib\distutils\util.py", line 291, in execute
    func(*args)
  File "C:\Users\User\.venv\lib\site-packages\setuptools\wheel.py", line 101, in install_as_egg
    self._install_as_egg(destination_eggdir, zf)
  File "C:\Users\User\.venv\lib\site-packages\setuptools\wheel.py", line 109, in _install_as_egg
    self._convert_metadata(zf, destination_eggdir, dist_info, egg_info)
  File "C:\Users\User\.venv\lib\site-packages\setuptools\wheel.py", line 131, in _convert_metadata
    zf.extractall(destination_eggdir)
  File "C:\Users\User\AppData\Local\Programs\Python\Python37-32\lib\zipfile.py", line 1619, in extractall
    self._extract_member(zipinfo, path, pwd)
  File "C:\Users\User\AppData\Local\Programs\Python\Python37-32\lib\zipfile.py", line 1674, in _extract_member
    shutil.copyfileobj(source, target)
  File "C:\Users\User\AppData\Local\Programs\Python\Python37-32\lib\shutil.py", line 79, in copyfileobj
    buf = fsrc.read(length)
  File "C:\Users\User\AppData\Local\Programs\Python\Python37-32\lib\zipfile.py", line 899, in read
    data = self._read1(n)
  File "C:\Users\User\AppData\Local\Programs\Python\Python37-32\lib\zipfile.py", line 975, in _read1
    data = self._decompressor.decompress(data, n)
zlib.error: Error -3 while decompressing data: invalid code lengths set
`

**Comment by bigFin** [comment]
Solved by installing scikit-image v 0.14.2, 
`pip install scikit-image==0.14.2`

looks like this error is a known bug in 0.14.3 [LINK](https://github.com/pyxem/pyxem/issues/422)

**Comment by nfahlgren** [comment]
Ah okay. We have been avoiding scikit-image 0.15 for now because they changed the canny edge detection function. Looks like pip is installing 0.14.3, which as you pointed to is not working on Windows for some reason. We did some testing and we will update `requirements.txt` to pin scikit-image to 0.14.2 for now. I confirmed that on Windows if you do the install with conda it is installing 0.14.2 by default also. Thanks for bringing this to our attention!
